AirCasting/CreateSessionViews/TurnOnLocationView.swift

```diff
@@ -0,0 +1,108 @@
+// Created by Lunar on 27/07/2021.
+//
+
+import AirCastingStyling
+import CoreBluetooth
+import SwiftUI
+
+struct TurnOnLocationView: View {
+    @State private var isPowerABLinkActive = false
+    @State private var showAlert = false
+    @State private var isTurnBluetoothOnLinkActive = false
+    @State private var isMobileLinkActive = false
+    @EnvironmentObject var settingsRedirection: DefaultSettingsRedirection
+    @Binding var creatingSessionFlowContinues: Bool
+    @EnvironmentObject var bluetoothManager: BluetoothManager
+    @EnvironmentObject private var locationTracker: LocationTracker
+    @StateObject var sessionContext: CreateSessionContext
+    
+    let urlProvider: BaseURLProvider
+    
+    var body: some View {
+        VStack(spacing: 50) {
+            ProgressView(value: 0.125)
+            Image("location-1")
+            VStack(alignment: .leading, spacing: 15) {
+                titleLabel
+                messageLabel
+            }
+            continueButton
+                .buttonStyle(BlueButtonStyle())
+        }
+        .alert(isPresented: $showAlert) {
+            Alert.locationAlert
+        }
+        .background(
+            Group {
+                NavigationLink(
+                    destination: PowerABView(creatingSessionFlowContinues: $creatingSessionFlowContinues, urlProvider: urlProvider),
+                    isActive: $isPowerABLinkActive,
+                    label: {
+                        EmptyView()
+                    })
+                NavigationLink(
+                    destination: TurnOnBluetoothView(creatingSessionFlowContinues: $creatingSessionFlowContinues, sessionContext: sessionContext, urlProvider: urlProvider),
+                    isActive: $isTurnBluetoothOnLinkActive,
+                    label: {
+                        EmptyView()
+                    })
+                NavigationLink(
+                    destination: SelectDeviceView(creatingSessionFlowContinues: $creatingSessionFlowContinues, urlProvider: urlProvider),
+                    isActive: $isMobileLinkActive,
+                    label: {
+                        EmptyView()
+                    })
+            }
+        )
+        .padding()
+        .onAppear {
+            locationTracker.requestAuthorisation()
```

Comment:
this is a completely new view, I think we should have ViewModels by default now to handle logic

(please apply to whole view)